src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -448,18 +470,43 @@ function tryRemoveElement(elem) {
             destroyFlvPlayer(this);
             destroyCastPlayer(this);
 
+            let secondaryTrackValid = true;
+
             this.#subtitleTrackIndexToSetOnPlaying = options.mediaSource.DefaultSubtitleStreamIndex == null ? -1 : options.mediaSource.DefaultSubtitleStreamIndex;
             if (this.#subtitleTrackIndexToSetOnPlaying != null && this.#subtitleTrackIndexToSetOnPlaying >= 0) {
                 const initialSubtitleStream = options.mediaSource.MediaStreams[this.#subtitleTrackIndexToSetOnPlaying];
                 if (!initialSubtitleStream || initialSubtitleStream.DeliveryMethod === 'Encode') {
                     this.#subtitleTrackIndexToSetOnPlaying = -1;
+                    secondaryTrackValid = false;
+                }
+                // secondary track should not be shown if primary track is no longer a valid pair
+                if (initialSubtitleStream && !playbackManager.trackHasSecondarySubtitleSupport(initialSubtitleStream, this)) {
+                    secondaryTrackValid = false;
                 }
+            } else {
+                secondaryTrackValid = false;
             }
 
             this.#audioTrackIndexToSetOnPlaying = options.playMethod === 'Transcode' ? null : options.mediaSource.DefaultAudioStreamIndex;
 
             this._currentPlayOptions = options;
 
+            // Get the secondary track that has been set during this watch session
+            let currentSecondaryTrackIndex = playbackManager.getSecondarySubtitleStreamIndex(this);
+
+            if (!secondaryTrackValid) {
+                currentSecondaryTrackIndex = -1;
+                playbackManager.setSecondarySubtitleStreamIndex(currentSecondaryTrackIndex, this);
+            }
+
+            this.#secondarySubtitleTrackIndexToSetOnPlaying = currentSecondaryTrackIndex == null ? -1 : currentSecondaryTrackIndex;
+            if (this.#secondarySubtitleTrackIndexToSetOnPlaying != null && this.#secondarySubtitleTrackIndexToSetOnPlaying >= 0) {
+                const initialSecondarySubtitleStream = options.mediaSource.MediaStreams[this.#secondarySubtitleTrackIndexToSetOnPlaying];
+                if (!initialSecondarySubtitleStream || initialSecondarySubtitleStream.DeliveryMethod !== 'External') {
+                    this.#secondarySubtitleTrackIndexToSetOnPlaying = -1;
+                }
+            }
```

Comment:
Using `getSecondarySubtitleStreamIndex` has a side effect:
1. Run video with 2 SRTs.
2. Select both subtitles (primary and secondary).
3. Stop playing.
4. Run the same video.
5. Secondary subtitles are displayed, but they are not marked in the menu because they are cleared in `onPlaybackStarted`. :spaghetti: :man_facepalming: 

If we want to reset the secondary subtitle selection when we start a new session, we can do the following:
```suggestion
            if (secondaryTrackValid) {
                this.#secondarySubtitleTrackIndexToSetOnPlaying = options.mediaSource.DefaultSecondarySubtitleStreamIndex == null ? -1 : options.mediaSource.DefaultSecondarySubtitleStreamIndex;
                if (this.#secondarySubtitleTrackIndexToSetOnPlaying != null && this.#secondarySubtitleTrackIndexToSetOnPlaying >= 0) {
                    const initialSecondarySubtitleStream = options.mediaSource.MediaStreams[this.#secondarySubtitleTrackIndexToSetOnPlaying];
                    if (!initialSecondarySubtitleStream || initialSecondarySubtitleStream.DeliveryMethod !== 'External') {
                        this.#secondarySubtitleTrackIndexToSetOnPlaying = -1;
                        playbackManager.setSecondarySubtitleStreamIndex(-1, this);
                    }
                }
            } else {
                this.#secondarySubtitleTrackIndexToSetOnPlaying = -1;
                playbackManager.setSecondarySubtitleStreamIndex(-1, this);
            }
```
`playbackManager.setSecondarySubtitleStreamIndex(-1, this)` can probably be removed if we remove this condition:
https://github.com/jellyfin/jellyfin-web/blob/3c59efaf700622384483bd581100ae748bcbf435/src/plugins/htmlVideoPlayer/plugin.js#L901
But I am not sure if this will cause problems with playbackmanager (race).

Reply:
Okay, so I tried with removing the `playbackManager.setSecondarySubtitleStreamIndex(-1, this)` and it seems to be working fine. I have not run into anything strange. 

It is now functioning as I mentioned before. It will always reset the secondary subtitle to a `null/undefined` value by default. The secondary subtitle track index will only be carried over if we are going to the next/prev track from the player itself.

I think this is the best approach for now until some sort of official user secondary subtitle default language setting can be implemented.